hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/AsyncSnapshotWriter.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.util;
+
+import com.hazelcast.nio.serialization.Data;
+
+import javax.annotation.CheckReturnValue;
+import java.util.Map.Entry;
+
+public interface AsyncSnapshotWriter {
+    void setCurrentMap(String mapName);
+
+    @CheckReturnValue
+    boolean offer(Entry<? extends Data, ? extends Data> entry);
+
+    @CheckReturnValue
+    boolean flushRemaining();
+
+    /**
+     * @throws Throwable Any error occurred during writing to underlying map.
+     * Error is reported only once, next call will say whether the pending flushes are done.
+     */
+    boolean hasPendingFlushes() throws Throwable;
```

Comment:
maybe the name `isReady()` is better. Error checking also could be separated into another method such as `hasErrors()`.

Reply:
fixed